### ISSUE TITLE
Problems implementing CEDC

### ISSUE THREAD

**Issue by Phronesis255** [AnFreTh/STREAM#101]
I'd really like to try out the CEDC algorithm you presented in Topics in a Haystack, but the documentation is unclear as to how embeddings or tokenized docs can be turned into a TMDataset.

**Comment by AnFreTh** [comment]
Hi, there is an example how to load a custom dataset here:
https://github.com/AnFreTh/STREAM/blob/main/docs/notebooks/datasets.ipynb

I hope this helps :)
If it doesnt, feel free to submit a minimal code example such that we can recreate the issue.

**Comment by Phronesis255** [comment]
Thank you, it helped a lot! At least I got it to produce the output. I'm just starting out, so my questions may be uh not exactly bright :D

Despite getting most of it right, I ran into an error when trying to visualize the topics. Here's my entire code. I'm experimenting at this stage, getting the top 30 results for a keyword and saving them to a csv with a random label (A, B, C, D, or E). It seems I'm doing something wrong because I get: "Cannot use scipy.linalg.eigh for sparse A with k >= N. Use scipy.linalg.eigh(A.toarray()) or reduce k."

my code
```
lemmatized_docs=pd.read_csv("lemmatized_frame.csv")
dataset = TMDataset()

dataset.create_load_save_dataset(
    data=lemmatized_docs, 
    dataset_name="sample_data",
    save_dir="data/",
    doc_column="Documents",
    label_column="Labels"
)
dataset.fetch_dataset(name="sample_data", dataset_path="data/", source="local")

dataset.preprocess(model_type="CEDC")


warnings.filterwarnings("ignore")
model = CEDC()
output = model.fit(dataset, n_topics=2)
print("passed this point")

from stream_topic.visuals import visualize_topic_model

#This is where I run into problems:

visualize_topic_model(
    model,
    reduce_first=True,
    port=8052,
)
```

**Comment by Phronesis255** [comment]
I'd be fine with any visualization, even a bar chart or a table would help me out at this point. Here is the csv I'm loading into it rn.

If you tell me how I can turn  the output from "output = model.fit(dataset, n_topics=3)" into a table, I'd be super grateful.

[lemmatized_frame.csv](https://github.com/user-attachments/files/17950320/lemmatized_frame.csv)
:

**Comment by Phronesis255** [comment]
@AnFreTh I think I managed to solve the problem by gathering more data. Is there a way to export the list of terms in each topic from the Dash visualization or the output object?

**Comment by AnFreTh** [comment]
Perfect. How to get topics, visualization etc. is all documented here: https://stream-topic.readthedocs.io/en/latest/
And also described in the readme.

For simplicty:
```python
topics = model.get_topics()
print(topics)
```

**Comment by Phronesis255** [comment]
Thank you for being patient with me. I'm reading the docs you linked above and they're great. I think I know what I'm doing now, so thanks again!
